### PR TITLE
[WIP] Gazebo11 update gtest (fix for windows build)

### PR DESCRIPTION
**Pull request by j-rivero**
The current build in Windows is failing with problems with gtest, staring with [this build](https://build.osrfoundation.org/job/gazebo-ci-gazebo11-windows7-amd64/10/). Problems started after merging [ UCRT headers](https://bitbucket.org/osrf/gazebo/pull-requests/3202/ensure-that-the-ucrt-headers-are-included/diff).

This PR updates gtest to latest version 1.10. `INSTANTIATE_TEST_CASE_P` is deprecated and `INSTANTIATE_TEST_SUITE_P` is the new name. That required also to tweaks part of our calls since the macro expansion is not happy with the line breaks and other artifacts.le

### CODE REVIEW

**Comment by j-rivero** [comment]
@osrf-jenkins run tests please

File: examples/stand_alone/test_fixture/gtest/src/gtest.cc [hunk 1]
```diff
@@ -2283,7 +2283,7 @@ class TestNameIs {
 namespace internal {
 
 // This method expands all parameterized tests registered with macros TEST_P
-// and INSTANTIATE_TEST_CASE_P into regular tests and registers those.
+// and INSTANTIATE_TEST_SUITE_P into regular tests and registers those.
```
Comment (scpeters) [inline examples/stand_alone/test_fixture/gtest/src/gtest.cc:2286]: this looks like a 2nd fork of gtest; I would revert this line change

File: gazebo/physics/Atmosphere_TEST.cc [hunk 2]
```diff
@@ -167,8 +167,8 @@ TEST_P(AtmosphereTest, AtmosphereParamBool)
   AtmosphereParamBool(this->GetParam());
 }
 
-INSTANTIATE_TEST_CASE_P(Atmospheres, AtmosphereTest,
-                        ::testing::Values("adiabatic"),);  // NOLINT
```
Comment (scpeters) [inline gazebo/physics/Atmosphere_TEST.cc:171]: I wonder if we can remote `NOLINT` now that the `,` isn't needed

**Comment by j-rivero** [comment]
a different approach in #2745

**Pull request closed without merging**